proposals/NNNN-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,239 @@
+* Status: **Awaiting review**
+* Implementation: [apple/allow-let-property-wrapper](https://github.com/apple/swift/pull/62342) 
+
+## Introduction
+
+[SE-0258 Property Wrappers](https://github.com/apple/swift-evolution/blob/main/proposals/0258-property-wrappers.md) models both mutable and immutable variants of a property wrapper but does not allow wrapped properties explicitly marked as a `let`. This proposal builds on the original vision by expanding property wrapper usage to include let declared properties with a wrapper attribute.
+
+## Motivation
+
+Allowing property wrappers to be applied to `let` properties improves Swift language consistency and expressivity.
+
+Today, a property wrapper can be applied to a property of any type but the rules for declaring these wrapped properties are varied and singular to property wrappers. For example, a struct instance without the property wrapper attribute can be declared with either a `var` or a `let` property. However, mark the struct as a property wrapper type and the compiler no longer allows it to be written as a `let` wrapped property:
+```swift
+@propertyWrapper
+struct Wrapper {
+  var wrappedValue: Int { 0 }
+  init(wrappedValue: Int) {}
+}
+
+struct S {
+  @Wrapper let value: Int // error: property wrapper can only be applied to a ‘var’
+}
+```
+Permitting wrapped properties to mimic the rules for other type instances that can be written with either a `var` or a `let` will simplify the Swift language. Additionally, `let` wrapped properties would expand property wrapper usage to allow `nonisolated` wrapped properties and simplify property wrappers usage where the backing type is a reference type.  
+
+
+## Proposed solution
+
+We propose to allow the application of property wrappers to let declared properties, which will permit the wrapper type to be initialized only once without affecting the implementation of the backing storage.
+
+For example, the following is the implementation for a `@BoilingPoint` property wrapper:
+```swift
+@propertyWrapper
+struct BoilingPoint<T> {
+  init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+
+  var wrappedValue: T
+  var projectedValue: T
+  
+  func toFarenheit() {}
+}
+
+struct Temperature {
+  @BoilingPoint var water: Double = 373.1
+}
+```
+
+The initial value for `water` is set using the `init(wrappedValue:)` initializer of the `BoilingPoint` property wrapper type. To ensure that `water` is not changed again, we could add logic to the `wrappedValue` setter to check if the property was already initialized and prevent re-assignment. However, this is an inconvenient solution.
+
+Instead, we could declare the `water` property with a `let`, synthesize a local `let` constant for the backing storage property (prefixed with an underscore), and only allow the property wrapper to be initialized once, passing the assigned value to `init(wrappedValue:)`. Now, rewriting `Temperature`’s properties as `let` constants will translate to:
```

Comment:
```suggestion
Instead, we declare the `water` property with `let`. The compiler synthesizes a local `let` declaration for the backing storage property, passing the assigned value to `init(wrappedValue:)`:

```

Not _could_ (a hypothetical): this proposal is about exactly allow this for real. This section needs to make clear what the _user_ is writing and what's synthesized.